src/transform.c

```diff
@@ -822,12 +822,17 @@ int transform_load(struct augeas *aug, struct tree *xfm) {
         // FIXME: Record an error and return 0
         return -1;
     }
+
     r = filter_generate(xfm, aug->root, &nmatches, &matches);
     if (r == -1)
         return -1;
     for (int i=0; i < nmatches; i++) {
         const char *filename = matches[i] + strlen(aug->root) - 1;
         struct tree *finfo = file_info(aug, filename);
+
+        if (file != NULL && STREQ(filename, file) != 0)
```

Comment:
I got the logic in my initial comment wrong; this line should read
```c
if (file != NULL && STRNEQ(filename, file))
```

That's the main reason why the test did not pass.